packages/headless/src/controllers/commerce/core/result-list/headless-core-interactive-result.ts

```diff
@@ -1,28 +1,41 @@
 import {Product} from '../../../../api/commerce/common/product';
 import {CommerceEngine} from '../../../../app/commerce-engine/commerce-engine';
+import {productClick} from '../../../../features/commerce/context/product/product-actions';
 import {
   buildInteractiveResultCore,
   InteractiveResultCore,
   InteractiveResultCoreOptions,
-  InteractiveResultCoreProps,
+  InteractiveResultCoreProps as InteractiveResultHeadlessCoreProps,
 } from '../../../core/interactive-result/headless-core-interactive-result';
 
 export interface InteractiveResultOptions extends InteractiveResultCoreOptions {
   /**
    * The product.
    */
+  // TODO: Which Product should this be? The headless one or the relay one? The product view controller uses the relay one.
```

Comment:
I personally think it would make more sense for headless controllers to expose the headless `Product` and not the relay `Product`. My understanding is that we can map relay `Product` properties from a headless `Product` (as I've done below).

The TODOs below have related questions as well.

@samisayegh @sallainCoveo thoughts?

Reply:
Hmm, since the products one would pass to the interactive result controller would normally have the shape of those returned by commerce api/headless, that suggests the headless `Product` interface.

However, this would hide errors because the product fields need to be present. We know this will sometimes not be the case because e.g. `ec_price` is fetched from a third-party system in real-time.

So, if we used the headless `Product` interface, at best the fields would be optional e.g.

```
HProduct {
  ec_price?: number;
}
```
but `ec_price` is required for the analytics event

```
RProduct {
  ec_price: number;
}
```

Using the relay type would be more strict/accurate. But, it means that the following would error:

```
buildInteractiveResult({
  product: commerceEngine.products[0] // error: ec_price is optional but is required
}));
```
So one would have to do:

```
buildInteractiveResult({
  product: {
    ec_price: commerceEngine.products[0].ec_price!
    ... same for other fields ...
  }
})
```

tl;dr, I think the relay type would provide the more accurate type check, but it means implementers would need to map the fields, which is not super nice

Reply:
> We know this will sometimes not be the case because e.g.`ec_price` is fetched from a third-party system in real-time.

For this reason, I'd be comfortable exposing the relay `Product`. We must provide the flexibility because there is no way to consistently extract the product click payload from the headless product. I'll go with this option and then open the PR for review.